build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactmetrics/CollectMetricsTask.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.gradle.plugins.artifactmetrics
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.jvm.tasks.Jar
+import org.gradle.kotlin.dsl.getByName
+
+internal abstract class CollectMetricsTask : DefaultTask() {
+    @get:OutputFile
+    abstract val metricsFile: RegularFileProperty
+
+    init {
+        metricsFile.convention(project.layout.buildDirectory.file(OUTPUT_PATH + "artifactSizeMetrics.csv"))
+    }
+
+    @TaskAction
+    fun generateMetrics() {
+        val pluginConfig = this.project.rootProject.extensions.getByType(ArtifactMetricsConfig::class.java)
+        if (!project.path.startsWith(pluginConfig.artifactPrefixes)) return
```

Comment:
nit: prefer using stdlib functions over writing your own, I think this can be `if (pluginConfig.artifactPrefixes.none { it.startsWith(project.path) }) return`